### PR TITLE
[Text Analytics] Added entity linking tasks to analyze operation

### PR DESCRIPTION
**Pull request by suhas92**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeBatchActionsOperation.cs [hunk 1]
```diff
@@ -262,7 +262,7 @@ private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToke
                     _actionSucceeded = update.Value.Tasks.Completed;
                     _totalActions = update.Value.Tasks.Total;
 
-                    if (update.Value.Status == TextAnalyticsOperationStatus.Succeeded || update.Value.Status == TextAnalyticsOperationStatus.PartiallySucceeded)
+                    if (update.Value.Status == TextAnalyticsOperationStatus.Succeeded || update.Value.Status == TextAnalyticsOperationStatus.PartiallySucceeded || update.Value.Status == TextAnalyticsOperationStatus.PartiallyCompleted)
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeBatchActionsOperation.cs:265]: oh interesting, for curiosity, what is the difference between `PartiallySucceeded ` and `PartiallyCompleted`?

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeBatchActionsResult.cs [hunk 2]
```diff
@@ -43,6 +50,11 @@ internal AnalyzeBatchActionsResult(AnalyzeJobState jobState, IDictionary<string,
         /// </summary>
         public IReadOnlyCollection<RecognizePiiEntitiesActionResult> RecognizePiiEntitiesActionsResults { get; }
 
+        /// <summary>
+        /// Determines the collection of RecognizePiiEntitiesActionResult.
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeBatchActionsResult.cs:54]:
```suggestion
        /// Determines the collection of RecognizeLinkedEntitiesActionsResult.
```

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/EntityLinkingTaskParameters.cs [hunk 3]
```diff
@@ -11,15 +11,5 @@ namespace Azure.AI.TextAnalytics
     /// </summary>
     [CodeGenModel("EntityLinkingTaskParameters")]
     internal partial class EntityLinkingTaskParameters
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/EntityLinkingTaskParameters.cs:13]: could u also move this class to the `.Models` namespace?

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs [hunk 4]
```diff
@@ -2362,6 +2362,10 @@ private AnalyzeBatchActionsOperation StartAnalyzeBatchActions(MultiLanguageBatch
             {
                 tasks.KeyPhraseExtractionTasks = Transforms.ConvertFromKeyPhrasesOptionsToTasks(actions.ExtractKeyPhrasesOptions);
             }
+            if (actions.RecognizeLinkedEntitiesOptions != null)
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:2365]: the docstrings of the method (same with sync) need to be updated too to include entity linking. something like `multiple actions including entity recognition, PII entity recognition, linked entity recognition and key phrases extraction`.

**Review by maririos — COMMENTED** [review]
- Description [here](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/textanalytics/Azure.AI.TextAnalytics#run-analyze-operation-asynchronously) needs to be updated too 
- I don't see samples updated. Is the idea to do it in another PR?

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeBatchActionsOperation.cs (hunk 1 above)
Comment (suhas92) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeBatchActionsOperation.cs:265]: this is a duplicate, service has unified it to PartiallyCompleted.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeBatchActionsOperation.cs [hunk 5]
```diff
@@ -262,7 +262,8 @@ private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToke
                     _actionSucceeded = update.Value.Tasks.Completed;
                     _totalActions = update.Value.Tasks.Total;
 
-                    if (update.Value.Status == TextAnalyticsOperationStatus.Succeeded || update.Value.Status == TextAnalyticsOperationStatus.PartiallyCompleted)
+                    // TODO - Remove PartiallySucceeded once service deploys this to WestUS2
```
Comment (suhas92) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeBatchActionsOperation.cs:265]: Added this TODO, to remove `update.Value.Status == TextAnalyticsOperationStatus.PartiallySucceeded` on Friday.

File: sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,9 @@
 ## 5.1.0-beta.5 (Unreleased)
 ### New features
 - Added ability to filter the categories returned in a Personally Identifiable Information recognition with the optional parameter `CategoriesFilter` in `RecognizePiiEntitiesOptions`.
+- Added the ability to recognize linked entities under `AnalyzeBatchActionsOperation`.
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md:6]:
```suggestion
- Added the ability to recognize linked entities under `StartAnalyzeBatchActions`.
```

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeBatchActionsOperation.cs (hunk 5 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeBatchActionsOperation.cs:265]: create an issue also to track it in case we forget

**Review by maririos — APPROVED** [review]

**Pull request merged**